packages/nsid/src/index.ts

```diff
@@ -0,0 +1,70 @@
+/*
+Grammar:
+
+alpha     = "a" / "b" / "c" / "d" / "e" / "f" / "g" / "h" / "i" / "j" / "k" / "l" / "m" / "n" / "o" / "p" / "q" / "r" / "s" / "t" / "u" / "v" / "w" / "x" / "y" / "z" / "A" / "B" / "C" / "D" / "E" / "F" / "G" / "H" / "I" / "J" / "K" / "L" / "M" / "N" / "O" / "P" / "Q" / "R" / "S" / "T" / "U" / "V" / "W" / "X" / "Y" / "Z"
+number    = "1" / "2" / "3" / "4" / "5" / "6" / "7" / "8" / "9" / "0"
+delim     = "."
+segment   = alpha *( alpha / number / "-" )
+authority = segment *( delim segment )
+name      = segment
+nsid      = authority delim name
+nsid-ns   = authority delim "*"
+
+*/
+
+const SEGMENT_RE = /^[a-zA-Z]([a-zA-Z0-9-])*$/
+
+export class NSID {
+  segments: string[] = []
+
+  constructor(nsid: string) {
+    const segments = nsid.split('.')
+    if (segments.length <= 1) {
```

Comment:
shouldn't this need to be `< 3`?
Since an authority needs to have 2 parts?

Reply:
Yeah that would make sense, I don't imagine anybody should be using TLDs as their authority